client/src/partials/templates/modals/confirmDeletion.modal.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+.controller('ConfirmDeletionModalController', ConfirmDeletionModalController);
+
+// dependencies injection
+ConfirmDeletionModalController.$inject = ['$uibModalInstance', '$translate', 'data'];
+
+/**
+ * Confirm Deletion Controller
+ * This controller is responsible for check a match text given to continue
+ * the deletion process
+ */
+function ConfirmDeletionModalController (Instance, $translate, Data) {
+  var vm = this;
+
+  // Global objects
+  vm.pattern = Data.pattern;
+  vm.elementName = Data.elementName;
+
+  // expose functions to the view
+  vm.close = close;
+  vm.accept = accept;
+
+  // startup
+  startup();
```

Comment:
Since the `startup()` method won't need to be called multiple times, you do not really need this method.